climada/engine/unsequa/calc_delta_climate.py

```diff
@@ -0,0 +1,457 @@
+        haz_initial_input_var: Union[InputVar, Hazard],
+        exp_final_input_var: Union[InputVar, Exposures],
+        impf_final_input_var: Union[InputVar, ImpactFuncSet],
+        haz_final_input_var: Union[InputVar, Hazard],
+    ):
+        """Initialize UncCalcImpact
+
+        Sets the uncertainty input variables, the impact metric_names, and the
+        units.
+
+        Parameters
+        ----------
+        exp_initial_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.Exposure
+            Exposure uncertainty variable or Exposure
+        impf_initital_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.ImpactFuncSet
+            Impact function set uncertainty variable or Impact function set
+        haz_initial_input_var : climada.engine.uncertainty.input_var.InputVar or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard
+        exp_final_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.Exposure
+            Exposure uncertainty variable or Exposure
+        impf_final_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.ImpactFuncSet
+            Impact function set uncertainty variable or Impact function set
+        haz_final_input_var : climada.engine.uncertainty.input_var.InputVar or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard
+
+        """
+
+        Calc.__init__(self)
+        self.exp_initial_input_var =  InputVar.var_to_inputvar(exp_initial_input_var)
+        self.impf_initial_input_var =  InputVar.var_to_inputvar(impf_initial_input_var)
+        self.haz_initial_input_var =  InputVar.var_to_inputvar(haz_initial_input_var)
+        self.exp_final_input_var =  InputVar.var_to_inputvar(exp_final_input_var)
+        self.impf_final_input_var =  InputVar.var_to_inputvar(impf_final_input_var)
+        self.haz_final_input_var =  InputVar.var_to_inputvar(haz_final_input_var)
+
+        self.value_unit = self.exp_initial_input_var.evaluate().value_unit
+        self.check_distr()
+
+
+    def uncertainty(self,
+                    unc_sample,
+                    rp=None,
+                    calc_eai_exp=False,
+                    calc_at_event=False,
+                    processes=1,
+                    chunksize=None
+                    ):
+        """
+        Computes the differential impact between the reference (initial) and 
+        future (final)for each sample in unc_data.sample_df.
+
+        By default, the aggregated average impact within a period of 1/frequency_unit
+        (impact.aai_agg) and the excees impact at return periods rp
+        (imppact.calc_freq_curve(self.rp).impact) is computed.
+        Optionally, eai_exp and at_event is computed (this may require
+        a larger amount of memory if the number of samples and/or the number
+        of centroids and/or exposures points is large).
```

Comment:
Maybe here one should explain a bit better what exactly are the values reported. Is it the difference of the outputs such as `eai_exp`, or is the difference first done on the impact matrix and then the outputs are computed? This is in particular would make a difference for the return periods.

Reply:
okay, I added a specific description incl. example.